test/spmd/test_dynamo_spmd.py

```diff
@@ -85,6 +85,92 @@ def test_dynamo_spmd_output_sharding_cache(self):
     dynamo_res = dynamo_linear(xla_y)
     self.assertEqual(met.counter_value('UncachedOutputSharding'), 1)
 
+  def test_dynamo_sharded_input(self):
+    device = xm.xla_device()
+    linear = SimpleLinear().to(device)
+    linear.eval()
+    xla_x = torch.randn(8, 128, device=device)
+    xs.mark_sharding(xla_x, self._get_mesh((1, self.n_devices)), (1, 0))
+    xla_res = linear(xla_x)
+    xm.mark_step()
+
+    dynamo_linear = torch.compile(linear, backend="openxla")
+    dynamo_res = dynamo_linear(xla_x)
+    torch.allclose(xla_res.cpu(), dynamo_res.cpu())
+
+  def test_dynamo_input_sharding_changed(self):
+    device = xm.xla_device()
+    linear = SimpleLinear().to(device)
+    linear.eval()
+    xla_x = torch.randn(8, 128, device=device)
+    xla_y = torch.randn(8, 128, device=device)
+    xm.mark_step()
+
+    met.clear_all()
+    dynamo_linear = torch.compile(linear, backend="openxla")
+    dynamo_res = dynamo_linear(xla_x)
+    self.assertIn('CompileTime', met.metric_names())
+    self.assertEqual(met.metric_data('CompileTime')[0], 1)
+
+    # Shard the original input
+    xs.mark_sharding(xla_x, self._get_mesh((1, self.n_devices)), (1, 0))
+    dynamo_res_sharded = dynamo_linear(xla_x)
+    torch.allclose(dynamo_res.cpu(), dynamo_res_sharded.cpu())
+    # one graph is being generated by .cpu call above
```

Comment:
Should .cpu() trigger another compilation? Maybe a separate topic but I'm curious. I assume it shouldn't?

Reply:
I am guessing the graph is something similar to 
```
%1 = device_data()
```
didn't check the actual HLO